includes/multi-currency/class-frontend-currencies.php

```diff
@@ -0,0 +1,155 @@
+<?php
+/**
+ * WooCommerce Payments Multi Currency Frontend Currencies
+ *
+ * @package WooCommerce\Payments
+ */
+
+namespace WCPay\Multi_Currency;
+
+defined( 'ABSPATH' ) || exit;
+
+/**
+ * Class that formats Multi Currency currencies on the frontend.
+ */
+class Frontend_Currencies {
+	/**
+	 * Multi-Currency instance.
+	 *
+	 * @var Multi_Currency
+	 */
+	protected $multi_currency;
+
+	/**
+	 * Multi-Currency currency formatting map.
+	 *
+	 * @var array
+	 */
+	protected $currency_settings = [];
+
+	/**
+	 * Constructor.
+	 *
+	 * @param Multi_Currency $multi_currency The Multi_Currency instance.
+	 */
+	public function __construct( Multi_Currency $multi_currency ) {
+		$this->multi_currency = $multi_currency;
+
+		$this->load_locale_data();
+
+		// Currency hooks.
+		add_filter( 'woocommerce_currency', [ $this, 'get_current_currency_code' ] );
+		add_filter( 'wc_get_price_decimals', [ $this, 'get_current_currency_decimals' ] );
+		add_filter( 'wc_get_price_decimal_separator', [ $this, 'get_current_currency_decimal_separator' ] );
+		add_filter( 'wc_get_price_thousand_separator', [ $this, 'get_current_currency_thousand_separator' ] );
+		add_filter( 'woocommerce_price_format', [ $this, 'get_current_currency_format' ] );
```

Comment:
Should we register these hooks with a higher priority to ensure they're executed after the default priority?

Reply:
That makes sense to me even if just to mitigate the risk of weird edge-case issues.

Reply:
👍  Added in https://github.com/Automattic/woocommerce-payments/pull/1860/commits/950f6095631fdd12d953cc7d09b7353a042ccb1d